script/build.ts

```diff
@@ -131,21 +131,23 @@ function packageApp() {
     )
   }
 
-  const toPackageArch = (targetArch: string | undefined): OfficialArch => {
-    if (targetArch === undefined) {
-      targetArch = os.arch()
-    }
+  const getPackageArch = (): OfficialArch => {
+    const arch = process.env.npm_config_arch || process.arch
 
     if (
-      targetArch === 'arm64' ||
-      targetArch === 'x64' ||
-      targetArch === 'armv7l'
+      arch === 'arm64' ||
+      arch === 'x64' ||
+      arch === 'armv7l'
```

Comment:
@theofficialgman are you saying that I shouldn't be returning `armv71` here, and should remove this case? That seems to be present in the original PR, and I've pushed 6d89d2705f2ee418ec184636173f59233d77bd86 which tries to align this to match the other functions we've updated in here...

Reply:
see my new comment https://github.com/shiftkey/desktop/pull/898#discussion_r1257539165

Reply:
for some reason process.arch does not work here. I am not sure why but I had build errors when I put it in this file particularly

Reply:
ah haha, didn't realize that was the issue from the error messages https://github.com/shiftkey/desktop/pull/898/commits/6e47e944610114ae734018750abe60f540e6cff0

Reply:
> @theofficialgman are you saying that I shouldn't be returning `armv71` here, and should remove this case? That seems to be present in the original PR, and I've pushed [6d89d27](https://github.com/shiftkey/desktop/commit/6d89d2705f2ee418ec184636173f59233d77bd86) which tries to align this to match the other functions we've updated in here...

to answer your question more directly. `armv7l` needs to be returned. but the input of this function will never be `armv7l`. os.arch/process.arch/env.npm_config_arch will always be `arm`. so the case you have
```
 ||
      arch === 'armv7l'
```
is really unnecessary since it will never be hit, and if it is, thats not good because that means the user incorrectly put `env.npm_config_arch` to armv7l which won't correctly compile for arm

Reply:
Taking a shot at this with 3e8e6cc6dfe91922f96905cb5e6353b2b2caa3ef but I think we could unify all these functions - will follow-up after now that we know these are lurking here...